arangod/Aql/AstNode.cpp

```diff
@@ -1768,7 +1769,8 @@ bool AstNode::canRunOnDBServer() const {
   if (type == NODE_TYPE_FCALL) {
     // built-in function
     auto func = static_cast<Function*>(getData());
-    if (func->hasFlag(Function::Flags::CanRunOnDBServer)) {
+    if ((isOneShard && func->hasFlag(Function::Flags::CanRunOnDBServerOneShard)) ||
+        (!isOneShard && func->hasFlag(Function::Flags::CanRunOnDBServerCluster))) {
```

Comment:
Can we have the case that `isOneShard == true`, `func->hasFlag(Flags::CanRunOnDBServerCluster) == true` but `func->hasFlag(Flags::CanRunOnDBServerOneShard) == false`? In that case we would not allow the function to run on the server, even though it could.

Reply:
The flags are currently (intentionally) separated, so they can be set independently of each other.
Currently there is no use case for this, but there may be one in the future.

As currently the intended behavior is that OneShard implies Cluster, I only added some assertions that validate it here and when a Function instance is created.